std/fmt.zig

```diff
@@ -256,126 +397,111 @@ pub fn formatType(
 fn formatValue(
     value: var,
     comptime fmt: []const u8,
+    comptime options: FormatOptions,
     context: var,
     comptime Errors: type,
     output: fn (@typeOf(context), []const u8) Errors!void,
 ) Errors!void {
-    if (fmt.len > 0 and fmt[0] == 'B') {
-        comptime var width: ?usize = null;
-        if (fmt.len > 1) {
-            if (fmt[1] == 'i') {
-                if (fmt.len > 2) {
-                    width = comptime (parseUnsigned(usize, fmt[2..], 10) catch unreachable);
-                }
-                return formatBytes(value, width, 1024, context, Errors, output);
-            }
-            width = comptime (parseUnsigned(usize, fmt[1..], 10) catch unreachable);
-        }
-        return formatBytes(value, width, 1000, context, Errors, output);
+    if (comptime std.mem.eql(u8, fmt, "B")) {
```

Comment:
I've changed all the format specifier checks to check the entirety of the format string and not simply the first character. Previously you could do something like `{sqpweouqwe}` and it would be valid, provided the first few happened to be valid.